### ISSUE TITLE
Deprecate Python 2 and Odoo < 12.0. New versioning for project

### ISSUE THREAD

**Issue by antonag32** [OCA/pylint-odoo#381]
As part of a modernization effort, which includes issues like #267 and #380 we should use a new versioning schema which will allow us to deprecate old versions of both Python and Odoo, this will let us remove dead code and improve the project.

After speaking with @moylop260 our first goal is to deprecate Python 2 and Odoo versions < 12.0, this will let us remove checks like
utf-8 coding comment (required for Python < 3) or `missing-import-error` [which is only active on Odoo <= 11](https://github.com/OCA/pylint-odoo/blob/1c91fce3127021a5ff8f81d9c54314eb612057be/pylint_odoo/checkers/modules_odoo.py#L276)

Right now the pip version is 6.2.0. We can freeze this branch, stop adding new features and only fix bugs if necessary, following semantic versioning, new versions would be 6.2.1, 6.2.2, etc...

For all the breaking changes we are about to propose we should increase a major version, so the new release of pylint-odoo will start at 7.0.0. I think we should base ourselves on the version management style used by pylint, this means:

1. Our `master` branch should be a sort of "rolling release". We continually work on it and push new releases by linking them to a specific commit in the branch. [Like this commit in pylint.](https://github.com/PyCQA/pylint/commit/680edebc686cad664bbed934a490aeafa775f163)
2. If we ever need to make a breaking change or push a new major release, we create a new branch which we will "freeze", this branch wont include the breaking chnage. For example, to keep maintaining version `6.2.0`, we will create a branch `6.2` on github. Fixes for 6.2.0 will be pushed to that branch, while new development will continue in `master`.

In short pylint-odoo 7.0 will take us to the moon :rocket: :rocket: 

Feedback is encouraged and appreciated, thanks.
